ams/server/src/main/java/com/netease/arctic/server/optimizing/maintainer/TableMaintainer.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netease.arctic.server.optimizing.maintainer;
+
+import com.netease.arctic.AmoroTable;
+import com.netease.arctic.ams.api.TableFormat;
+import com.netease.arctic.server.table.TableRuntime;
+import com.netease.arctic.table.ArcticTable;
+import org.apache.iceberg.Table;
+
+/**
+ * API for maintaining table.
+ * <p> Includes: clean content files, clean metadata, clean dangling delete files, expire snapshots.
+ */
+public interface TableMaintainer {
+
+  /**
+   * Clean content files includes data files, delete files.
+   */
+  void cleanContentFiles(long lastTime);
```

Comment:
I'm considering whether it's appropriate to put the parameters that an operation depends on in the interface parameters. The specific parameters for a certain operation may undergo significant changes with version updates. Should we instead put these parameters in the constructor of the specific implementation class?

Reply:
There is a problem with putting all operations' parameters in the constructor, as the user may only use one of the functionalities and not all of the operations' parameters. If other parameters are not passed, it may cause confusion for users who do not necessarily know how the creator constructed it.